src/store/actions.ts

```diff
@@ -31,8 +32,20 @@ export default ( searchEntityRepository: SearchEntityRepository, metricsCollecto
 		context.commit( 'setValue', payload );
 	},
 	updateProperty( context: ActionContext<RootState, RootState>,
-		payload: { property: Property; conditionIndex: number } ): void {
+		payload: { property: { label: string; id: string; datatype: string }; conditionIndex: number } ): void {
+
 		context.commit( 'setProperty', payload );
+		if ( payload.property && !allowedDatatypes.includes( payload.property.datatype ) ) {
+			context.dispatch( 'setConditionAsLimitedSupport', payload.conditionIndex );
+		} else {
+			context.commit(
+				'setFieldErrors',
```

Comment:
For bonus points you could create a separate `clearFieldErrors` mutation instead of using `setFieldErrors` with `null`. I tripped over this line a tiny bit while reading through it.

I didn't manage to review this PR in detail but what I saw looked good! :)

Reply:
Done in #106 :)